peeratmac/rancid-tomatillos
# Testing unit hf

### What's this PR do?
 *Navigation Bar*
- Adds second snapshot to test rendering for logged in vs not logged in
- Adds Unit tests for invocation of two props methods
- Stills needs testing for mapStateToProps and mapDispatchToProps <*Done by @Dunn-Austen's PR*  
- *Unit test still needed for handleError prop*  

 *Login Form*
- Adds props to snapshot tests and adds 2 more snapshot tests to test conditional rendering
- Adds Unit tests for invocation of all functions and local state updates ***except*** those within a `.then()` or `.catch()` of a fetch.

 *Movie Container*
- Updates original snapshot and adds second snapshot to test conditional rendering
- Testing completed for this component (issue #23) 

*Movie Card*
- This PR doesn't add any tests to Movie Card but no unit tests needed so testing completed (issue #22 )

*MovieShowPage*
- Updates and adds snapshots for conditional rendering
- Adds commented out it blocks for the needed unit tests
- Commented out it blocks include questions on a few I'm not 100% sure on

### Where should the reviewer start?
- Go into each component noted, look at the tests, confirm if anything is missing from test suite, check out commented out code for any struggle points/questions.

### How should this be manually tested?
- Pull down and run test suite.  
- See added snapshot tests and how they differ.
- Review unit test suite to ensure full testing coverage
- See commented code in each component's test suite to see questions/testing that hasn't yet been completed.

### What are the relevant tickets?
- Works on issue #25  and issue #21 and issue #26 
- issue #22 and issue #23 should be completed 

### Is there any background information to provide?
- You'll see commits from @Dunn-Austen's open branch in this PR as well because I pulled it down and merged it into my branch to ensure I wasn't re-writing any tests.

### What's next on this PR?
- Actually writing the tests for the MovieShowPage 
- Revisiting tests that were not successfully completed in Navigation Bar and Login Form
